app/models/interview_question.rb

```diff
@@ -2,4 +2,27 @@ class InterviewQuestion < ActiveRecord::Base
   belongs_to :interview
   validates :interview_id, presence: true
   validates :question_id, presence: true
+  validates :mark, presence: true, on: :update
+
+  def self.get_interview_question_number(interview, interview_question)
+    interview_question_number = 1
+    interview.interview_questions.each_with_index do |q, i|
+      interview_question_number = i + 1 if q.question_id == interview_question.question_id
+    end
+    interview_question_number
+  end
+
+  def self.color_link(question)
+    color = "bg_link"
+    color = "bg_link bg_green" if question.mark == 'green'
+    color = "bg_link bg_yellow" if question.mark == 'yellow'
+    color = "bg_link bg_red" if question.mark == 'red'
+    color
```

Comment:
This is definitely belongs to view layer and not to model.
Also a lot of redundancy in this code.